File: backend/app/models/llm_user_config.py
"""
LLM 用户配置模型
存储用户的模型配置信息
"""
from sqlalchemy import Column, Integer, String, Boolean, Float, Text, DateTime, func
from app.core.database import Base


class LLMUserConfig(Base):
    """用户 LLM 配置表"""
    __tablename__ = "llm_user_configs"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    
    # 用户标识（可以是 user_id、session_id 或 device_id）
    user_identifier = Column(String(100), nullable=False, index=True, comment="用户标识")
    
    # LLM 配置
    provider = Column(String(50), nullable=False, default="custom", comment="Provider 类型")
    model = Column(String(100), nullable=False, comment="模型名称")
    api_key = Column(Text, nullable=True, comment="API Key（加密存储）")
    base_url = Column(Text, nullable=True, comment="Base URL")
    auth_type = Column(String(20), nullable=False, default="bearer", comment="认证类型: bearer, token, api_key, custom")
    auth_header = Column(String(50), nullable=True, comment="自定义认证头名称")
    api_format = Column(String(50), nullable=False, default="openai", comment="API 格式: openai, anthropic, gemini")
    is_full_url = Column(Boolean, nullable=False, default=False, comment="是否使用完整 URL")
    
    # 高级配置
    temperature = Column(Float, nullable=False, default=0.3, comment="温度参数")
    max_tokens = Column(Integer, nullable=False, default=2048, comment="最大 token 数")
    thinking = Column(Boolean, nullable=False, default=False, comment="是否启用思考模式")
    max_input_tokens = Column(Integer, nullable=True, default=180000, comment="最大输入 token 数")
    
    # 元数据
    is_active = Column(Boolean, nullable=False, default=True, comment="是否为当前激活配置")
    config_name = Column(String(100), nullable=True, comment="配置名称（可选）")
    
    # 时间戳
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), comment="更新时间")
    last_used_at = Column(DateTime(timezone=True), nullable=True, comment="最后使用时间")
    
    def to_dict(self):
        """转换为字典（不包含敏感信息）"""
        return {
            "id": self.id,
            "user_identifier": self.user_identifier,
            "provider": self.provider,
            "model": self.model,
            # api_key 不返回，需要时单独获取
            "base_url": self.base_url,
            "auth_type": self.auth_type,
            "auth_header": self.auth_header,
            "api_format": self.api_format,
            "is_full_url": self.is_full_url,
            "temperature": self.temperature,
            "max_tokens": self.max_tokens,
            "thinking": self.thinking,
            "max_input_tokens": self.max_input_tokens,
            "is_active": self.is_active,
            "config_name": self.config_name,
            "created_at": self.created_at.isoformat() if self.created_at else None,
            "updated_at": self.updated_at.isoformat() if self.updated_at else None,
            "last_used_at": self.last_used_at.isoformat() if self.last_used_at else None,
        }
    
    def to_full_dict(self):
        """转换为完整字典（包含敏感信息）"""
        data = self.to_dict()
        data["api_key"] = self.api_key
        return data

File: backend-app/src/main/java/com/sitech/prodai/domain/entity/LlmUserConfig.java
package com.sitech.prodai.domain.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * 用户 LLM 配置 —— 对齐 Python {@code app/models/llm_user_config.py::LLMUserConfig}。
 */
@Getter
@Setter
@NoArgsConstructor
@TableName("pd_ai_llm_user_configs")
@Schema(description = "LLM 模型配置（pd_ai_llm_user_configs）：推理平台接入参数与运行时调优项")
public class LlmUserConfig {

    @TableId(type = IdType.AUTO)
    @Schema(description = "配置 ID（新增时不传，更新时必传）")
    private Integer id;

    @TableField("user_identifier")
    @Schema(description = "所属用户标识")
    private String userIdentifier;

    @TableField("provider")
    @Schema(description = "模型提供方：openai / azure / custom / local", example = "custom")
    private String provider = ModelProvider.CUSTOM.getValue();

    @TableField("model")
    @Schema(description = "模型名称", example = "deepseek-chat")
    private String model;

    @TableField("api_key")
    @Schema(description = "API 密钥（部分本地服务可空）")
    private String apiKey;

    @TableField("base_url")
    @Schema(description = "服务基础地址", example = "https://api.deepseek.com/v1")
    private String baseUrl;

    @TableField("auth_type")
    @Schema(description = "认证方式：bearer / header")
    private String authType = "bearer";

    @TableField("auth_header")
    @Schema(description = "自定义认证头名称（auth_type=header 时生效）")
    private String authHeader;

    @TableField("api_format")
    @Schema(description = "API 协议格式：openai 兼容等")
    private String apiFormat = "openai";

    @TableField("is_full_url")
    @Schema(description = "base_url 是否已含 /v1 等完整路径：true 后端仅追加 /chat/completions")
    private Boolean isFullUrl = false;

    @TableField("temperature")
    @Schema(description = "采样温度（0~2，越大越随机）", example = "0.3")
    private Double temperature = 0.3;

    @TableField("max_tokens")
    @Schema(description = "单次生成最大 token 数", example = "2048")
    private Integer maxTokens = 2048;

    @TableField("thinking")
    @Schema(description = "是否开启深度思考模式（需模型支持）")
    private Boolean thinking = false;

    @TableField("stream_enabled")
    @Schema(description = "是否启用流式输出（SSE）")
    private Boolean streamEnabled = true;

    @TableField("max_input_tokens")
    @Schema(description = "输入上下文最大 token 数（超限自动截断）", example = "180000")
    private Integer maxInputTokens = 180000;

    @TableField("is_active")
    @Schema(description = "是否为当前全局生效配置（唯一激活）")
    private Boolean isActive = true;

    @TableField("config_name")
    @Schema(description = "配置显示名称", example = "DeepSeek 生产环境")
    private String configName;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private LocalDateTime updatedAt;

    @TableField("last_used_at")
    @Schema(description = "最近使用时间")
    private LocalDateTime lastUsedAt;
}
